### PR TITLE
arm_kvm_unit_tests: Run kvm-unit-tests subtests and check results

### PR DESCRIPTION
**Pull request by PaulYuuu**
Clone the kvm-unit-tests and run specific subtests, then check the test results.

ID: 2176735, 2176734
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
Test result:
```
 (1/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.its.arm64-pci: STARTED
 (1/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.its.arm64-pci: PASS (64.15 s)
 (2/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.gic.arm64-pci: STARTED
 (2/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.gic.arm64-pci: PASS (39.43 s)
 (3/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.pmu.arm64-pci: STARTED
 (3/3) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.pmu.arm64-pci: FAIL: Certain pmu test cases fail: ['pmu-chained-counters', 'pmu-chained-sw-incr', 'pmu-chain-promotion', 'pmu-overflow-interrupt'] (52.86 s)
```

File: generic/tests/cfg/arm_kvm_unit_tests.cfg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+- arm_kvm_unit_tests:
+    virt_test_type = qemu libvirt
+    type = arm_kvm_unit_tests
+    only aarch64
+    vms = ""
+    start_vm = no
+    repo_url = https://gitlab.com/kvm-unit-tests/kvm-unit-tests
+    variants sub_type:
```
Comment (zhenyzha) [inline generic/tests/cfg/arm_kvm_unit_tests.cfg:8]: Do we need to split it? Just like `xfstest`, we can directly run this collection `./run_tests.sh`, and then print out the error cases.

File: generic/tests/cfg/arm_kvm_unit_tests.cfg [hunk 2]
```diff
@@ -0,0 +1,11 @@
+- arm_kvm_unit_tests:
+    virt_test_type = qemu libvirt
+    type = arm_kvm_unit_tests
```
[the remark this answers is not in the document]
Comment (PaulYuuu) [reply generic/tests/cfg/arm_kvm_unit_tests.cfg:3]: gic, its, pmu are only for aarch64. And tp-qemu already have a very old kvm_unit_test scenario.

File: generic/tests/cfg/arm_kvm_unit_tests.cfg (hunk 1 above)
Comment (PaulYuuu) [reply generic/tests/cfg/arm_kvm_unit_tests.cfg:8]: Actually I only want to run sub set in a sub test, this is more controllable for fine-grained tests. But I will use `--group` for the testing.

Comment (PaulYuuu) [reply generic/tests/cfg/arm_kvm_unit_tests.cfg:8]: Hum, `--group` always returns `0`, I need to consider again.

File: generic/tests/arm_kvm_unit_tests.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    for test_file in glob(os.path.join(tests_dir, sub_type + '*')):
+        test_name = os.path.basename(test_file)
+        s, o = process.getstatusoutput(test_file)
+        test.log.debug(f'Output of "{test_name}":\n{o}')
+        if s and s != 77:
+            failed_tests.append(os.path.basename(test_file))
+
+    if failed_tests:
+        test.fail(f'Certain {sub_type} test cases fail: {failed_tests}')
+    test.log.info(f'All {sub_type} tests passed')
```
[the remark this answers is not in the document]
Comment (PaulYuuu) [reply generic/tests/arm_kvm_unit_tests.py]: Remove the repo already.

File: generic/tests/cfg/arm_kvm_unit_tests.cfg (hunk 1 above)
Comment (PaulYuuu) [reply generic/tests/cfg/arm_kvm_unit_tests.cfg:8]: Finally, I want to keep this, it's more controllable for subgroups.

File: generic/tests/arm_kvm_unit_tests.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    try:
+        error_context.base_context(f'Run {sub_type} sub tests', test.log.info)
+        process.system(f'cd {repo_dir} && ./configure && make standalone',
+                       verbose=False, shell=True)
+        for test_file in glob(os.path.join(tests_dir, sub_type + '*')):
+            test_name = os.path.basename(test_file)
+            s, o = process.getstatusoutput(test_file)
+            test.log.debug(f'Output of "{test_name}":\n{o}')
+            if s and s != 77:
```
Comment (zhenyzha) [inline generic/tests/arm_kvm_unit_tests.py:43]: Sorry, I didn't understand this step.

**Comment by zhenyzha** [comment]
The other part is ok for me.

git am 3637.patch
Applying: arm_kvm_unit_tests: Run kvm-unit-tests subtests and check results

```
 (1/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.its.arm64-pci: STARTED
 (1/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.its.arm64-pci: PASS (41.80 s)
 (2/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.gic.arm64-pci: STARTED
 (2/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.gic.arm64-pci: PASS (34.77 s)
 (3/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.pmu.arm64-pci: STARTED
 (3/3) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.arm_kvm_unit_tests.pmu.arm64-pci: PASS (38.83 s)
```

**Review by zhenyzha — APPROVED** [review]

**Pull request merged**